src/components/tasks/card/SuggestionBox.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { FC, useEffect, useState } from 'react';
+import { DUMMY_PROFILE as placeholderImageURL } from '@/constants/display-sections';
+import classNames from '@/components/tasks/card/suggestion.module.scss';
+import fetch from '@/helperFunctions/fetch';
+import { userDataType } from '@/interfaces/user.type';
+import { GithubInfo } from '@/interfaces/suggestionBox.type';
+
+type Props = {
+    assigneeName: string;
+    showSuggestion: boolean;
+    onClickName: (userName: string) => void;
+};
+
+const SuggestionBox: FC<Props> = ({
+    assigneeName,
+    showSuggestion = false,
+    onClickName = () => undefined,
+}) => {
+    const [loading, setLoading] = useState<boolean>(false);
+    const [suggestions, setSuggestions] = useState<GithubInfo[]>([]);
+
+    const clickHandler = (e: React.MouseEvent<HTMLLIElement, MouseEvent>) => {
+        onClickName(e.currentTarget.innerText);
+    };
+
+    const fetchUsers = async () => {
+        setLoading(true);
+        const url = `${process.env.NEXT_PUBLIC_BASE_URL}/users?search=${assigneeName}`;
+        const { requestPromise } = fetch({ url });
+        const users = await requestPromise;
+        const usersData = users.data.users;
+        const suggestedUsers: GithubInfo[] = [];
+        usersData.map((data: userDataType) => {
+            suggestedUsers.push({
+                github_id: data.github_id,
+                profileImageUrl: data?.picture?.url
+                    ? data.picture.url
+                    : placeholderImageURL,
+            });
+        });
+
+        setSuggestions(suggestedUsers);
+        setLoading(false);
+    };
+
+    useEffect(() => {
+        let timerId: NodeJS.Timeout;
+        if (assigneeName) {
```

Comment:
what are we trying to do here ?

Reply:
So, i do not want to send request on every keystroke, so i just introduced a timeout, now if user is continuously typing I do not want to send the request, and hence i am storing the timerId, so that i can clean up that particular setTimeOut out of the queue. I hope this makes sense @vvaibhavdesai

Reply:
I think a better way to do this would be using useCallback rather than using a useEffect